bundles/statistics/statsgrid2016/components/IndicatorParameters.js

```diff
@@ -139,8 +140,18 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.IndicatorParameters', function(
                     }
                     me.service.getStateService().setRegionset(regionSelect.value());
 
-                    // FIXME need this anymore ?
-                    //me.instance.getFlyout().closePanels();
+                    var flyoutManager = (me.instance && me.instance.getTile() && me.instance.getTile().getFlyoutManager()) ? me.instance.getTile().getFlyoutManager() : null;
+                    var searchFlyout = (flyoutManager && flyoutManager.getFlyout('search')) ? flyoutManager.getFlyout('search') : null;
```

Comment:
You can take out the last ifs to make it more readable as the check for searchFlyout will basically be the same if the value is undefined or null. I would consider refactoring this to make it work without going through so many references. Like using some internal event through the service to trigger a flyout to open or some other way.

Reply:
Is this really something that should be the responsibility of indicator parameters in the first place? Maybe you should just call a method in extrafeatures that an indicator was added. It should decice what to do with the information, not this component.